TestApi/TestApi/Controllers/AllocationController.cs

```diff
@@ -87,16 +60,18 @@ public async Task<IActionResult> AllocateUserByUserTypeAndApplicationAsync(UserT
         /// </summary>
         /// <param name="request">Allocate users request</param>
         /// <returns>Full details of an allocated users</returns>
-        [HttpPut("allocateUsers")]
+        [HttpGet("allocateUsers")]
         [ProducesResponseType(typeof(List<UserDetailsResponse>), (int)HttpStatusCode.OK)]
         [ProducesResponseType((int)HttpStatusCode.BadRequest)]
         public async Task<IActionResult> AllocateUsersAsync(AllocateUsersRequest request)
         {
             _logger.LogDebug($"AllocateUsersAsync No. of UserTypes: {request.UserTypes.Count} Application: {request.Application}");
 
-            if (request.UserTypes.Count.Equals(0))
+            var result = await new AllocateUsersRequestValidator().ValidateAsync(request);
```

Comment:
Have a look at the ValidationFactory we have in bookings api/video api. It automatically executes a validation for a dto if one exists withuot you having to call it specifically in a controller

Reply:
I copied it from the booking-api

Reply:
<img width="597" alt="image" src="https://user-images.githubusercontent.com/10450246/90131608-b3bdfa80-dd6c-11ea-872a-c8b8cf65843d.png">

This is from bookings-api ^^ is there a better way to do it?

Reply:
Done. Copied from video api as suggested.